docs/Modern-GraphQLInRelay.md

```diff
@@ -89,11 +89,68 @@ fragment TodoApp_app on App {
 }
 ```
 
+### `@inline`
+
+By default Relay will only expose the data for fields explicitly requested by a [component's fragment](./fragment-container.html#createfragmentcontainer), which is known as [data masking](./thinking-in-relay#data-masking). Fragment data is unmasked for use in React components by `createFragmentContainer`. However, you may want to use fragment data in non-React functions that are called from React.
+
+Non-React functions can also take advantage of data masking. A fragment can be defined with the `@inline` directive and stored in a local variable. The non-React function can then "unmask" the data using the `readInlineData` function.
+
+In the example below, the function `processItemData` is called from a React component. It requires an item object with a specific set of fields. All React components that use this function should spread the `processItemData_item` fragment to ensure all of the correct item data is loaded for this function. 
+
+```javascript
+import {graphql, readInlineData} from 'react-relay';
+
+// fragment for processItemData
+const processItemDataFragment = graphql`
+  fragment processItemData_item on Item @inline {
+    title
+    price
+    creator {
+      name
+    }
+  }
+`;
+
+// non-React function called from React
+function processItemData(itemRef) {
+  const item = readInlineData(processItemDataFragment, itemRef);
+  sendToThirdPartyApi({
+    title: item.title,
+    price: item.price,
+    creatorName: item.creator.name
+  });
+}
+```
+
+```javascript
+// React Component
```

Comment:
we should probably show this component's fragment too

Reply:
This component's fragment is a few lines down inside `createFragmentContainer`, did you have a different way of displaying it in mind?